### ISSUE TITLE
Game crashes when signing in

### ISSUE THREAD

**Issue by YoloSwaggings720** [playgameservices/play-games-plugin-for-unity#1935]
So I launch my game, press the sign in button and it crashes. Searched around and saw that there was problems with google play files version 10.2 and people resolved it by replacing it with 10.0.1, but now there is 10.2.6, so I don't think that's the problem.
I tried it with the sample app and it gives me the same crash with the same log error, so yea.

Here's the androidmanifest: 

    <manifest xmlns:android="http://schemas.android.com/apk/res/android"
        package="com.awsome.dunnn"
        android:versionCode="1"
        android:versionName="1.0" >

    <uses-sdk android:minSdkVersion="8" android:targetSdkVersion="16" />

    <application>

     
        <meta-data android:name="com.google.android.gms.nearby.connection.SERVICE_ID"
            android:value="" />

        
        <meta-data android:name="com.google.android.gms.games.APP_ID"
            android:value="\ 1***********1" />

        
        <meta-data android:name="com.google.android.gms.games.unityVersion"
            android:value="\ 0.9.41" />

        
        <meta-data android:name="com.google.android.gms.version"
            android:value="@integer/google_play_services_version" />
    </application>
</manifest>

**Here's the log from logcat:**

09-27 18:09:53.107: I/Unity(7999): clicked:SignIn
09-27 18:09:53.107: I/Unity(7999):  
09-27 18:09:53.107: I/Unity(7999): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)
09-27 18:09:53.152: D/GamesUnitySDK(7999): Performing Android initialization of the GPG SDK
09-27 18:09:53.240: I/Unity(7999): Building GPG services, implicitly attempts silent auth
09-27 18:09:53.240: I/Unity(7999):  
09-27 18:09:53.240: I/Unity(7999): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)
09-27 18:09:53.242: I/GamesNativeSDK(7999): Using existing jar.
09-27 18:09:53.242: I/GamesNativeSDK(7999): Writing 1941 bytes to jar file
09-27 18:09:53.252: I/GamesNativeSDK(7999): Using existing jar.
09-27 18:09:53.252: I/GamesNativeSDK(7999): Writing 1066 bytes to jar file
09-27 18:09:53.265: W/PopupManager(7999): You have not specified a View to use as content view for popups. Falling back to the Activity content view. Note that this may not work as expected in multi-screen environments
09-27 18:09:53.267: I/GamesNativeSDK(7999): Auth operation started: SIGN IN
09-27 18:09:53.267: I/GamesNativeSDK(7999): Connecting to Google Play...
09-27 18:09:53.294: W/GamesServiceBroker(21248): Client connected with SDK 10260000, Services 11517438, and Games 52250038
09-27 18:09:53.306: W/GamesServiceBroker(21248): Client connected with SDK 11411000, Services 11517438, and Games 52250038
09-27 18:09:53.310: E/ValidateServiceOp(21248): Using Google Play games services requires a metadata tag with the name "com.google.android.gms.games.APP_ID" in the application tag of the manifest for com.awsum.dunnn
09-27 18:09:53.315: D/AndroidRuntime(7999): Shutting down VM
09-27 18:09:53.318: E/AndroidRuntime(7999): FATAL EXCEPTION: main
09-27 18:09:53.318: E/AndroidRuntime(7999): Process: com.awsum.dunnn, PID: 7999
09-27 18:09:53.318: E/AndroidRuntime(7999): Theme: themes:{}
09-27 18:09:53.318: E/AndroidRuntime(7999): java.lang.Error: FATAL EXCEPTION [main]
09-27 18:09:53.318: E/AndroidRuntime(7999): Unity version     : 5.4.5p5
09-27 18:09:53.318: E/AndroidRuntime(7999): Device model      : OnePlus A0001
09-27 18:09:53.318: E/AndroidRuntime(7999): Device fingerprint: oneplus/bacon/A0001:6.0.1/MHC19Q/ZNH2KAS3P0:user/release-keys
09-27 18:09:53.318: E/AndroidRuntime(7999): Caused by: java.lang.IllegalStateException: A fatal developer error has occurred. Check the logs for further information.
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at com.google.android.gms.common.internal.zzf$zza.zzb(Unknown Source)
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at com.google.android.gms.common.internal.zzf$zza.zzu(Unknown Source)
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at com.google.android.gms.common.internal.zzf$zze.zzxH(Unknown Source)
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at com.google.android.gms.common.internal.zzf$zzd.handleMessage(Unknown Source)
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at android.os.Handler.dispatchMessage(Handler.java:102)
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at android.os.Looper.loop(Looper.java:148)
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at android.app.ActivityThread.main(ActivityThread.java:5461)
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at java.lang.reflect.Method.invoke(Native Method)
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
09-27 18:09:53.318: E/AndroidRuntime(7999): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
09-27 18:09:53.319: W/ActivityManager(806):   Force finishing activity com.awsum.dunnn/com.unity3d.player.UnityPlayerActivity
09-27 18:09:53.394: W/Unity(7999): !!! [Play Games Plugin DLL] 09/27/17 18:09:53 +03:00 WARNING: Creating new PlayGamesPlatform
09-27 18:09:53.394: W/Unity(7999):  
09-27 18:09:53.394: W/Unity(7999): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)
09-27 18:09:53.394: I/Unity(7999):  [Play Games Plugin DLL] 09/27/17 18:09:53 +03:00 DEBUG: Activating PlayGamesPlatform.
09-27 18:09:53.394: I/Unity(7999):  
09-27 18:09:53.394: I/Unity(7999): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)
09-27 18:09:53.394: I/Unity(7999):  [Play Games Plugin DLL] 09/27/17 18:09:53 +03:00 DEBUG: PlayGamesPlatform activated: GooglePlayGames.PlayGamesPlatform
09-27 18:09:53.394: I/Unity(7999):  
09-27 18:09:53.394: I/Unity(7999): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)
09-27 18:09:53.394: I/Unity(7999):  [Play Games Plugin DLL] 09/27/17 18:09:53 +03:00 DEBUG: Creating platform-specific Play Games client.
09-27 18:09:53.394: I/Unity(7999):  
09-27 18:09:53.394: I/Unity(7999): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)
09-27 18:09:53.395: I/Unity(7999):  [Play Games Plugin DLL] 09/27/17 18:09:53 +03:00 DEBUG: Creating Android IPlayGamesClient Client
09-27 18:09:53.395: I/Unity(7999):  
09-27 18:09:53.395: I/Unity(7999): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)
09-27 18:09:56.071: W/System(13918): ClassLoader referenced unknown path: /system/framework/tcmclient.jar
09-27 18:09:57.416: I/WindowState(806): WIN DEATH: Window{15fd070 u0 com.awsum.dunnn/com.unity3d.player.UnityPlayerActivity}
09-27 18:09:57.416: W/WindowManager(806): Force-removing child win Window{944b195 u0 SurfaceView} from container Window{15fd070 u0 com.awsum.dunnn/com.unity3d.player.UnityPlayerActivity}
09-27 18:09:57.416: D/ConnectivityService(806): releasing NetworkRequest NetworkRequest [ id=1549, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
09-27 18:09:57.417: E/ConnectivityService(806): RemoteException caught trying to send a callback msg for NetworkRequest [ id=1549, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
09-27 18:09:57.417: D/ConnectivityService(806): releasing NetworkRequest NetworkRequest [ id=1550, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
09-27 18:09:57.418: E/ConnectivityService(806): RemoteException caught trying to send a callback msg for NetworkRequest [ id=1550, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
09-27 18:09:57.419: W/WindowManager(806): Failed looking up window
09-27 18:09:57.419: W/WindowManager(806): java.lang.IllegalArgumentException: Requested window android.os.BinderProxy@dee0fe9 does not exist
09-27 18:09:57.419: W/WindowManager(806): 	at com.android.server.wm.WindowManagerService.windowForClientLocked(WindowManagerService.java:8802)
09-27 18:09:57.419: W/WindowManager(806): 	at com.android.server.wm.WindowManagerService.windowForClientLocked(WindowManagerService.java:8793)
09-27 18:09:57.419: W/WindowManager(806): 	at com.android.server.wm.WindowState$DeathRecipient.binderDied(WindowState.java:1212)
09-27 18:09:57.419: W/WindowManager(806): 	at android.os.BinderProxy.sendDeathNotice(Binder.java:558)
09-27 18:09:57.419: I/WindowState(806): WIN DEATH: null

**Comment by YoloSwaggings720** [comment]
I noticed that in my manifest file the package name was "com.google.example.games.mainlibproj". I changed it, but still the same.

I'm guessing that somewhere the package name is missing and now it auto-generated this package name and the service cant find the APP_ID tag.

**Comment by martasp** [comment]
Same problem authentification always returns false even in the empty fresh folder.

**Comment by supersolid-Justy** [comment]
Check in your Android manifest if:
"com.google.android.gms.games.APP_ID" 
APP_ID is populated with an actual id :)

**Comment by YoloSwaggings720** [comment]
Here is the manifest file and yes the id is there, but another problem is that at setup, Google Play had auto generated the manifest with it's example package name: **com.google.example.games.mainlibproj**.
I changed the package name, but still the same error. Here is the manifest:

`<?xml version="1.0" encoding="utf-8"?>

<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.awsome.dunnn"
    android:versionCode="1"
    android:versionName="1.0" >

    <uses-sdk android:minSdkVersion="8" android:targetSdkVersion="16" />

    <application>

     
        <meta-data android:name="com.google.android.gms.nearby.connection.SERVICE_ID"
            android:value="" />

        
        <meta-data android:name="com.google.android.gms.games.APP_ID"
            android:value="\ 1****************3" />

        
        <meta-data android:name="com.google.android.gms.games.unityVersion"
            android:value="\ 0.9.41" />

        
        <meta-data android:name="com.google.android.gms.version"
            android:value="@integer/google_play_services_version" />
    </application>
</manifest>`

**Comment by supersolid-Justy** [comment]
Hm..okay so in my test bed project i also have got the package package="com.google.example.games.mainlibproj" and it does work fine for me anyway. So I think you might have a problem with the libraries.
You have **play-services-games** **AND** **play-services-auth** on 10.2.6
Underneath I posted a pic of the files I have in my project. There are more of them than needed, as I just copied them from a different project as well, but it should just serve as an example.
Also did you run in Unity->Window->Google Play Games->Setup->Android Setup ----> Setup? The GameInfo file might not have been populated.

![untitled](https://user-images.githubusercontent.com/16538905/31009453-6e55cf3e-a4ff-11e7-8637-d3629d1bcea7.png)

**Comment by YoloSwaggings720** [comment]
The problem was(like the error said) that it couldn't find the APP_ID in the manifest. So I looked into the main manifest that got generated after you build the apk and guess what, the Google play manifest(in the assets folder underneath googleplaygames folder and under plugin android) was being completely ignored while the manifests where merged. So there was no APP_ID in the generated manifest, while it was in the Google play manifest. That's why it was so confusing.

So I overwrote the Unity manifest, by copying the previous generated manifest and adding the missing GooglePlay stuff in manually. See [this](https://docs.unity3d.com/Manual/android-manifest.html) for how to overwrite manifest. Keep in mind that I had Unity Ads manifest as well, so I had to leave that part out of my created manifest because later those 2 manifests would merge.

I can't understand why there was no Google Play Games developer support here suggesting to look into the main manifest. God damn 3 days for this bullshit.
